crates/wysiwyg/src/composer_model/hyperlinks.rs

```diff
@@ -58,9 +61,63 @@ where
 
     pub fn set_link(&mut self, link: S) -> ComposerUpdate<S> {
         self.push_state_to_history();
-        let (s, e) = self.safe_selection();
-        let range = self.state.dom.find_range(s, e);
-        self.set_link_range(range, link)
+        let (mut s, mut e) = self.safe_selection();
+
+        let mut range = self.state.dom.find_range(s, e);
+
+        // Find container link that completely covers the range
+        if let Some(link) = self.find_parent_links(&range) {
+            // If found, update the range to the container link bounds
+            range = self.state.dom.find_range_by_node(&link);
+            (s, e) = (range.start(), range.end());
+        }
```

Comment:
If there is already a parent link and we found it, couldn't we just modify it to update its link value instead? Or replace it with a copy that contains the same children but updated values.

I might be missing some corner case.

Reply:
That may be more efficient but I think it'd be safer (and simpler) to process this special case in the same way as other links (so that there is only one code path).

Reply:
I'd argue about it being simpler, it took me a while to understand why you were doing this. However, I think it's fine to leave it like this.